lib/Reader/readerir.cpp

```diff
@@ -1879,11 +1902,13 @@ bool GenIR::isUnmanagedPointerType(llvm::Type *Type) {
 //
 //===----------------------------------------------------------------------===//
 
-EHRegion *fgNodeGetRegion(FlowGraphNode *Node) { return nullptr; }
-
-EHRegion *fgNodeGetRegion(llvm::Function *Function) { return nullptr; }
+EHRegion *GenIR::fgNodeGetRegion(FlowGraphNode *Node) {
+  return FlowGraphInfoMap[Node].Region;
+}
 
-void fgNodeSetRegion(FlowGraphNode *Node, EHRegion *Region) { return; }
+void GenIR::fgNodeSetRegion(FlowGraphNode *Node, EHRegion *Region) {
+  FlowGraphInfoMap[Node].Region = Region;
```

Comment:
I tried your suggestion but it failed [here](https://github.com/JosephTremoulet/llilc/commit/62fd29482a3e9ade54b98b54dcb022a8df95c5d1#diff-f3d514eb35910c5b4b5e3c8b5af72317R2986):
```c++
      if (fgNodeGetStartMSILOffset(Block) != CurrentOffset) {
        // Split the block so we can maintain a many-to-one block-to-region
        // mapping
        Block = fgSplitBlock(Block, CurrentOffset, nullptr);
      }
      // Set the region on the current block
      fgNodeSetRegion(Block, CurrentRegion);
```
since I changed splitBlock to copy the region to the newly-created tail but then here I patch it up because we're splitting for the sake of starting a new region.  I'm inclined to just leave things alone, but I could add an explicit region parameter to splitBlock if you think it's important.